### PR TITLE
CHANGE docs for 0.6 release

### PR DESCRIPTION
**Pull request by ndelangen**
I've given the docs an overhaul. I'll work on it some more later I think.

Feedback is very welcome.
---

I spend a considerable amount of time getting the scrolling behavior to work in in casperjs; something I did made it not work anymore, and thus I refactored that a bit into what seemed like a sane test to me.

### CODE REVIEW

File: assets/css/base/buttons.less [hunk 1]
```diff
@@ -3,27 +3,27 @@
 
 /* TODO: cosmetic changes needed */
 .source_btn {
-	position:relative;
+    position:relative;
```
Comment (operatino) [inline assets/css/base/buttons.less]: I see different code style is applied here and there, from 4 to 2 spaces. Let's stick to one, and add csslint/eslint/editorconfig? Keeping the 4 spaces as the standard.

Comment (operatino) [inline assets/css/base/buttons.less]: And I would really like to have style changes as separate PR next time.

**Comment by operatino** [comment]
Getting an error on `/docs/data-nav/` page

```
[2016-06-18 17:00:41] [WARN] - Could not pre-render spec with EJS: /docs/data-nav/ [Error: Could not find matching close tag for "<%=".]
Error: Could not find matching close tag for "<%=".
    at /Users/robert-lg/LocalProjects/Source/node_modules/ejs/lib/ejs.js:550:19
    at Array.forEach (native)
    at Object.Template.generateSource (/Users/robert-lg/LocalProjects/Source/node_modules/ejs/lib/ejs.js:536:15)
    at Object.Template.compile (/Users/robert-lg/LocalProjects/Source/node_modules/ejs/lib/ejs.js:458:12)
    at Object.compile (/Users/robert-lg/LocalProjects/Source/node_modules/ejs/lib/ejs.js:318:16)
    at handleCache (/Users/robert-lg/LocalProjects/Source/node_modules/ejs/lib/ejs.js:177:16)
    at exports.render (/Users/robert-lg/LocalProjects/Source/node_modules/ejs/lib/ejs.js:345:10)
    at Object.ejs.render (/Users/robert-lg/LocalProjects/Source/core/ejsWithHelpers.js:147:12)
    at exports.process (/Users/robert-lg/LocalProjects/Source/core/middlewares/preRenderEjs.js:31:33)
    at Layer.handle [as handle_request] (/Users/robert-lg/LocalProjects/Source/node_modules/express/lib/router/layer.js:95:5)
```

File: docs/base/readme.md [hunk 2]
```diff
@@ -75,26 +73,35 @@ We treat Spec files as an interface, you can construct Spec page in many ways fo
...
-
 The starting template for new Spec pages can be found in `sourcejs/docs/starting` folder. Copy the contents to a new folder in `source/user/specs` and you'll be ready to write a new spec.
 
-<a href="/docs/spec" class="source_a_hl">Check the SourceJS Spec page documentation.</a>
+[Check the SourceJS Spec page documentation.](/docs/spec)
 
 ### Server-side Templating Engines
 
-As we mentioned before, it's easy to use other server-side templating engines like Jade, you only need to create a simple SourceJS middleware ([example](https://github.com/sourcejs/sourcejs-jade)) or process your sources with Grunt/Gulp.
+As we mentioned before, it's also possible to use *other* server-side templating engines like Jade. You need to create a SourceJS middleware that will perform the rendering. Our [sourcjejs-jade Middleware](https://github.com/sourcejs/sourcejs-jade) may serve as an example of such middleware.
```
Comment (operatino) [inline docs/base/readme.md]: Let's keep the name "plugin" for all "sourcejs-*" extensions, middleware is just a type of a plugin, and the name "plugin" in more approachable for non-node devs.

File: docs/base/readme.md [hunk 3]
```diff
@@ -116,58 +129,64 @@ We highly encourage you setting up Bootstrap bundle on your SourceJS instance, a
...
-Showcase includes both native Specs examples, and ones that are rendered with plugins like [sourcejs-contrib-dss](http://github.com/sourcejs/sourcejs-contrib-dss) and [sourcejs-jade](http://github.com/sourcejs/sourcejs-jade).
-
-<a href="/docs/spec" class="source_a_hl">Also check the SourceJS Spec page documentation.</a>
-
 ### Style Guide Driven Development
 
 To get more insights about recommended workflow within Style Guide platform check [this example bundle](https://www.youtube.com/watch?v=KeR8Qhgyb6M) together with short screencast.
 
-<div style="max-width: 600px;"><div style="position: relative; padding-bottom: 56.25%;">
+<div style="max-width: 100%;"><div style="position: relative; padding-bottom: 56.25%;">
```
Comment (operatino) [inline docs/base/readme.md]: looks too wide now, especially with poor thumbnail on youtube video

File: docs/base/readme.md [hunk 4]
```diff
@@ -116,58 +129,64 @@ We highly encourage you setting up Bootstrap bundle on your SourceJS instance, a
...
 <iframe width="100%" height="100%" style="position: absolute;" src="https://www.youtube.com/embed/KeR8Qhgyb6M" frameborder="0" allowfullscreen></iframe>
 </div></div>
 
 
 ## Configuration
 
-SourceJS engine is highly configurable and allows to override almost any options from your instance set-up using personal configuration file `sourcejs/user/options.js`. All default, and safe to change options are described in base configuration file `sourcejs/options.js`.
+SourceJS engine is highly configurable. Engine and Middleware are configured in 1 place:: `options.js`.
```
Comment (operatino) [inline docs/base/readme.md]: "Engine and plugins..."

File: docs/base/readme.md [hunk 5]
```diff
@@ -116,58 +129,64 @@ We highly encourage you setting up Bootstrap bundle on your SourceJS instance, a
...
 
 With version 0.5.3 we also introduced context level options `sourcejs-options.js`, which allows to configure any catalog specifically for your needs. Read more about it, and other configuration capabilities in [Engine Configuration](/docs/configuration) doc.
 
-## Plugins
+## Middleware
```
Comment (operatino) [inline docs/base/readme.md]: As mentioned above, I see "plugin"  more approachable term.

File: docs/base/readme.md [hunk 6]
```diff
@@ -116,58 +129,64 @@ We highly encourage you setting up Bootstrap bundle on your SourceJS instance, a
...
+* [sourcejs-spec-status](https://github.com/sourcejs/sourcejs-spec-status)
+* [sourcejs-crowd-voice](https://github.com/sourcejs/sourcejs-crowd-voice)
+* [sourcejs-specs-linting](https://github.com/sourcejs/sourcejs-specs-linting)
 
-These modules are able to extend both front-end and back-end part of the engine. To install any of official plugin, just use `npm install` in your `sourcejs/user` folder (note that some of them needs additional dependencies like [MongoDB](http://www.mongodb.org/) or [CouchDB](http://couchdb.apache.org/)).
+These modules are able to extend both front-end and back-end part of the engine. To install any middleware, run `npm install <name>`
+
+After install is finished (re)start your SourceJS instance using: `npm start`.
+
+Some Middleware need addition setup!
```
Comment (operatino) [inline docs/base/readme.md]: plugin

File: docs/base/readme.md [hunk 7]
```diff
@@ -116,58 +129,64 @@ We highly encourage you setting up Bootstrap bundle on your SourceJS instance, a
...
+* [sourcejs-dss](http://github.com/sourcejs/sourcejs-contrib-dss)
+
+### New example languages
+* [sourcejs-react-styleguidist](https://github.com/sourcejs/sourcejs-react-styleguidist)
+* [sourcejs-md-react](https://github.com/mik01aj/sourcejs-md-react)
+
+### New Utility
+* [sourcejs-browser-sync](https://github.com/sourcejs/sourcejs-contrib-browser-sync)
+
+### New Spec Parts
```
Comment (operatino) [inline docs/base/readme.md]: Let's make it "Spec extensions"

File: docs/base/readme.md [hunk 8]
```diff
@@ -116,58 +129,64 @@ We highly encourage you setting up Bootstrap bundle on your SourceJS instance, a
...
 Here is a list of available plugins:
 
-* [sourcejs-slm](https://github.com/venticco/sourcejs-slm) (new)
-* [sourcejs-md-react](https://github.com/mik01aj/sourcejs-md-react) (new)
-* [sourcejs-contrib-browser-sync](https://github.com/sourcejs/sourcejs-contrib-browser-sync) (new)
-* [sourcejs-react](https://github.com/szarouski/sourcejs-react) (new)
-* [sourcejs-contrib-dss](http://github.com/sourcejs/sourcejs-contrib-dss) (new)
-* [sourcejs-spec-status](https://github.com/sourcejs/sourcejs-spec-status)
-* [sourcejs-crowd-voice](https://github.com/sourcejs/sourcejs-crowd-voice)
+### New spec languages:
```
Comment (operatino) [inline docs/base/readme.md]: No need to use work "new", let just one section "Spec Language Integrations".  Newest could be just placed on top, and marked as bold.

Also, linking to npm search would be useful https://www.npmjs.com/search?q=sourcejs-

File: docs/data-nav/info.json [hunk 9]
```diff
@@ -1,4 +1,5 @@
 {
-    "title": "Auto-generated Navigation",
-    "template": "doc"
+    "title": "Creating catalogs",
```
Comment (operatino) [inline docs/data-nav/info.json]: it's more about creating catalog pages, rather the catalogs themselves

**Comment by operatino** [comment]
Huge tanks for the help with docs, that's one one of the blockers before final release. The hardest part left now, is to determine the migration path, and re-check compatibility.

**Comment by ndelangen** [comment]
Thanks for the feedback, I'll work on it more this week.

Sorry for the whitespace commit, I noticed I had placed tabs everywhere, so I **had** to clean it up.

File: assets/css/base/buttons.less (hunk 1 above)
Comment (ndelangen) [inline assets/css/base/buttons.less]: I've tried to make it 4 spaces everywhere, in this particular code example it's 4 spaces.

Comment (ndelangen) [inline assets/css/base/buttons.less]: eslint is definitely a must do!

**Comment by ndelangen** [comment]
Merged 0.6.0-dev into this branch, if you have no further comments, it can be merged in?

**Comment by ndelangen** [comment]
@operatino We should pick this up and possibly merge? Let's release 0.6!

**Comment by operatino** [comment]
@ndelangen, yeah, let's finally do it. I've been cleaning up the issues, not just need to sit down and re-test everything.

**Comment by ndelangen** [comment]
😴

**Comment by ndelangen** [comment]
🙁

**Comment by ndelangen** [comment]
😇

**Comment by operatino** [comment]
I'll revisit the file name structure in docs folder, but the rest is fine now. Next weeks I'll have few free days, will start final preps before 0.6 release.

**Pull request merged**